packages/web/pages/shopping-cart.js

```diff
@@ -0,0 +1,221 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React from 'react';
+import styled, { css } from 'styled-components';
+import { useForm } from 'react-hook-form';
+
+import Link from 'next/link';
+import { useTranslation } from 'react-i18next';
+import { RectangularButton } from '../components/Button';
+import EmptyScreen from '../components/EmptyScreen';
+import { SectionTitle } from '../components/Common';
+import { CartItem } from '../components/ShoppingCart';
+import { TextField } from '../components/Form';
+import { useAuth, useModal, useShoppingCart } from '../hooks';
+import { formatMoney } from '../utils/helper';
+import { createServiceOrder } from '../services';
+import { toast } from '../components/Toast';
+
+const ShoppingCart = () => {
+	const { t } = useTranslation(['common']);
+	const form = useForm({ comment: '' });
+	const { items, totalPrice, updateItem, removeItem } = useShoppingCart();
+	const { openModal } = useModal();
+	const { user } = useAuth();
+
+	const handleSubmit = async (values) => {
+		if (!user.id) {
+			return openModal('login', {
+				message: t('common:signInToContinue'),
+			});
+		}
+
+		if (!user.operations.can_create_service_order) {
+			return openModal('needToCompleteTheRegistration', null, {
+				overlayClick: false,
+			});
+		}
+
+		const result = await createServiceOrder(
+			items.map((item) => ({ service_id: item.id, quantity: item.quantity })),
+			values.comment,
+		);
+
+		if (!result) {
+			return toast.error(
+				'Ocorreu um erro ao finalizar seu pedido. Tente novamente em instantes.',
+			);
+		}
```

Comment:
In this situation, I believe that the errors that can be returned are diverse, so I think it would be a good idea **to capture the errors** thrown by backend to display them in frontend

Reply:
Yeah my you're totally right, but remember that we'll do that in a later moment, so we don't have to worry about errors in services because the API methods will do the job